pkg/gsheets/sheet.go

```diff
@@ -0,0 +1,185 @@
+		"release-openshift-ocp-installer-e2e-openstack-serial-4.3":    "Serial OCP 4.3",
+		"release-openshift-ocp-installer-e2e-openstack-4.2":           "Parallel OCP 4.2",
+		"release-openshift-ocp-installer-e2e-openstack-serial-4.2":    "Serial OCP 4.2",
+		"release-openshift-origin-installer-e2e-openstack-4.2":        "Parallel Origin 4.2",
+		"release-openshift-origin-installer-e2e-openstack-serial-4.2": "Serial Origin 4.2",
+	}
+
+	sheetName, ok := innerMap[s.JobName]
+	if !ok {
+		log.Fatalf("Unknown job %v", s.JobName)
+	}
+
+	return sheetName
+}
+
+func (s *Sheet) getAllIDs() {
+	readRange := s.getName() + "!A7:A300"
+	resp, err := s.Client.service.Spreadsheets.Values.Get(s.Client.spreadsheetId, readRange).Do()
+	if err != nil {
+		log.Fatalf("Unable to retrieve data from sheet: %v", err)
+	}
+
+	if len(resp.Values) == 0 {
+		fmt.Println("No data found.")
+		return
+	}
+
+	for _, row := range resp.Values {
+		id, err := strconv.Atoi(fmt.Sprint(row[0]))
+		if err != nil {
+			id = 0
+		}
+		s.idxList = append(s.idxList, id)
+	}
+}
+
+// Get the index on which the job with the given id should be inserted
+func (s *Sheet) getIndex(id int) (int, bool) {
+	// Populate list of IDs from sheet
+	if len(s.idxList) == 0 {
+		s.getAllIDs()
+	}
+
+	i := 0
+	for ; i < len(s.idxList); i++ {
+		// s.idxList is sorted in descending order
+		if s.idxList[i] <= id {
+			break
+		}
+	}
+
+	exists := len(s.idxList) > 0 && s.idxList[i] == id
+
+	return i, exists
+}
+
+// Get ID of the most recent job in the spreadsheet
+func (s *Sheet) GetLatestId() int {
+	readRange := s.getName() + "!A7"
+	resp, err := s.Client.service.Spreadsheets.Values.Get(s.Client.spreadsheetId, readRange).Do()
+	if err != nil {
+		log.Fatalf("Unable to retrieve data from sheet: %v", err)
+	}
+
+	if len(resp.Values) == 0 || len(resp.Values[0]) == 0 {
+		return 0
+	}
+
+	data := fmt.Sprint(resp.Values[0][0])
+	id, err := strconv.Atoi(data)
+	if err != nil {
+		return 0
+	}
+
+	return id
+}
+
+func (s *Sheet) AddRow(j job.Job) {
+	sheetId := s.getID()
+
+	id, _ := strconv.Atoi(j.ID)
+	idx, exists := s.getIndex(id)
+
+	if !exists {
+		// Create a new row to save the report
+		idr := &sheets.InsertDimensionRequest{
+			Range: &sheets.DimensionRange{
+				Dimension:  "ROWS",
+				SheetId:    sheetId,
+				StartIndex: int64(idx + offset),
+				EndIndex:   int64(idx + offset + 1),
+			},
+			InheritFromBefore: false,
+		}
+
+		request := &sheets.BatchUpdateSpreadsheetRequest{
+			Requests: []*sheets.Request{
+				&sheets.Request{
+					InsertDimension: idr,
+				},
+			},
+		}
+
+		_, err := s.Client.service.Spreadsheets.BatchUpdate(s.Client.spreadsheetId, request).Context(context.Background()).Do()
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		// Insert the new id into s.idxList
+		s.idxList = append(s.idxList, 0)
+		copy(s.idxList[idx+1:], s.idxList[idx:])
+		s.idxList[idx] = id
+	}
+
+	pdr := &sheets.PasteDataRequest{
```

Comment:
wow, congrats for finding this

Reply:
Sadly, there's no other way to add rendered HTML into gsheets.